packages/graphiql/cypress/integration/init.spec.js

```diff
@@ -0,0 +1,53 @@
+
+const testQuery = `{
+longDescriptionType {
+  id
+  image
+  hasArgs
+  test {
+    id
+    isTest
+    __typename
+  }
+ }
+}`
+
+const mockSuccess = {
+  "data": {
+    "longDescriptionType": {
+      "id": "abc123",
+      "image": "/images/logo.svg",
+      "hasArgs": "{\"defaultValue\":\"test default value\"}",
+      "test": {
+        "id": "abc123",
+        "isTest": true,
+        "__typename": "Test"
+      }
+    }
+  }
+}
+
+
+describe('GraphiQL On Initialization', function() {
+  it('Renders without error', function() {
+    cy.visit(`/?query=${testQuery}`)
+  })
+
+   it('Executes a query that has the expected resut', function() {
+    cy.get('.execute-button').click()
```

Comment:
Using classnames is brittle; it's fine if we're guaranteeing that these classes will always be there and will always be used for this purpose, but I prefer the `data-cy` approach [detailed here](https://docs.cypress.io/guides/references/best-practices.html#Selecting-Elements).

I tend to write a helper for this; in `cypress/support/commands.js` add:

```js
Cypress.Commands.add("getCy", cyName => {
  return cy.get(`[data-cy=${cyName}]`);
});
```

Then this line would become:
```suggestion
    cy.getCy('execute-button').click()
```

relating to something like `<button data-cy="execute-button">`. Note: we should commit to some kind of naming convention for these cypress-addressable names. We should also consider stripping `data-cy` from the production bundle.

Reply:
I wanted to add to the markup in another pass. I know full well that using classes for e2e tests are problematic. My goal for this PR is to only add cypress harness, to not change the actual components/etc yet. We will do this in another pass. I think the better solution is to use ID's, which is something we should be doing already. But we will be getting rid of most of our css classes soon anyways with styled components.